src/__tests__/RESTController-test.js

```diff
@@ -468,7 +468,7 @@ describe('RESTController', () => {
     jest.spyOn(options, 'progress');
 
     RESTController.ajax('POST', 'files/upload.txt', {}, {}, options).then(({ response, status }) => {
-      expect(options.progress).toHaveBeenCalledWith(null);
+      expect(options.progress).toHaveBeenCalledWith(null, null, null, { type: 'upload' });
```

Comment:
@dplewis @ladigitale  probably the reason why I'm kinda against to moving it at the end. This could be fixed when doing major upgrade tho. :smile:

Reply:
I agree with you a general rule of thumb is that more than 2 parameters can be the rest can be replaced with an object. I don't know about a major release just for something like this.

Also can you improve the documentation with an example.

https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseFile.js#L236

Reply:
I mean when doing a major upgrade like v5, this could be included has one of the breaking change.

Yes, I'm thinking about that as well. Thanks. :smile: